src/math/vec2.cpp

```diff
@@ -0,0 +1,41 @@
+//
+// Created by ajer on 16.12.18.
+//
+
+#include "vec2.h"
+
+namespace Bald::Math {
+
+
+    [[nodiscard]] constexpr Vec2 &Vec2::operator+(const Vec2 &other) noexcept {
+
+        this->m_x += other.m_x;
+        this->m_y += other.m_y;
+
+        return *this;
+    }
+
+    [[nodiscard]] constexpr Vec2 &Vec2::operator*(double multiplier) noexcept {
+
+        if (multiplier != 0.0) {
```

Comment:
why cant't we multiply by 0? and we should use 0.0f notation

Reply:
done